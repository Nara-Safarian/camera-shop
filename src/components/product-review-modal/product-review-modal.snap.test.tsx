import { render } from '@testing-library/react';
import ProductReviewModal from './product-review-modal';
import { configureMockStore } from '@jedmao/redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureMockStore();

test('ProductReviewModal snapshot should be rendered correctly', () => {
  const {container} = render(
    <Provider store={mockStore({})}>
      <ProductReviewModal isActive onClose={function (): void {
        // nothing
      } } onSubmit={function (): void {
        // nothing
      } } cameraId='1'
      />
    </Provider>
  );
  expect(container).toMatchSnapshot();
});
