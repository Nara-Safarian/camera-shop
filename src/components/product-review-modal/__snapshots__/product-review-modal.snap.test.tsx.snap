// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductReviewModal snapshot should be rendered correctly 1`] = `
<div>
  <div
    class="visually-hidden"
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
    >
      <symbol
        id="icon-snowflake"
        viewBox="0 0 9 9"
      >
        <path
          d="M5.375 0.84375L5.05469 3.71875L7.96875 2.90625L8.22656 4.875L5.57031 5.0625L7.3125 7.38281L5.53906 8.32812L4.32031 5.88281L3.25 8.3125L1.40625 7.38281L3.13281 5.0625L0.492188 4.85938L0.796875 2.90625L3.64844 3.71875L3.32812 0.84375H5.375Z"
          fill="#ED6041"
        />
      </symbol>
      <symbol
        id="icon-close"
        viewBox="0 0 10 10"
      >
        <path
          clip-rule="evenodd"
          d="M1.70711 0.292893C1.31658 -0.0976311 0.683418 -0.0976311 0.292893 0.292893C-0.0976311 0.683418 -0.0976311 1.31658 0.292893 1.70711L3.58579 5L0.292893 8.29289C-0.0976311 8.68342 -0.0976311 9.31658 0.292893 9.70711C0.683418 10.0976 1.31658 10.0976 1.70711 9.70711L5 6.41421L8.29289 9.70711C8.68342 10.0976 9.31658 10.0976 9.70711 9.70711C10.0976 9.31658 10.0976 8.68342 9.70711 8.29289L6.41421 5L9.70711 1.70711C10.0976 1.31658 10.0976 0.683418 9.70711 0.292893C9.31658 -0.0976311 8.68342 -0.0976311 8.29289 0.292893L5 3.58579L1.70711 0.292893Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </symbol>
    </svg>
  </div>
  <div
    class="modal is-active"
  >
    <div
      class="modal__wrapper"
    >
      <div
        class="modal__overlay"
      />
      <div
        class="modal__content"
      >
        <p
          class="title title--h4"
        >
          Оставить отзыв
        </p>
        <div
          class="form-review"
        >
          <form
            method="post"
            novalidate=""
          >
            <div
              class="form-review__rate"
            >
              <fieldset
                class="rate form-review__item "
              >
                <legend
                  class="rate__caption"
                >
                  Рейтинг
                  <svg
                    aria-hidden="true"
                    height="9"
                    width="9"
                  >
                    <use
                      xlink:href="#icon-snowflake"
                    />
                  </svg>
                </legend>
                <div
                  class="rate__bar"
                >
                  <div
                    class="rate__group"
                  >
                    <input
                      class="visually-hidden"
                      id="star-5"
                      name="rate"
                      type="radio"
                      value="5"
                    />
                    <label
                      class="rate__label"
                      for="star-5"
                      title="Отлично"
                    />
                    <input
                      class="visually-hidden"
                      id="star-4"
                      name="rate"
                      type="radio"
                      value="4"
                    />
                    <label
                      class="rate__label"
                      for="star-4"
                      title="Хорошо"
                    />
                    <input
                      class="visually-hidden"
                      id="star-3"
                      name="rate"
                      type="radio"
                      value="3"
                    />
                    <label
                      class="rate__label"
                      for="star-3"
                      title="Нормально"
                    />
                    <input
                      class="visually-hidden"
                      id="star-2"
                      name="rate"
                      type="radio"
                      value="2"
                    />
                    <label
                      class="rate__label"
                      for="star-2"
                      title="Плохо"
                    />
                    <input
                      class="visually-hidden"
                      id="star-1"
                      name="rate"
                      type="radio"
                      value="1"
                    />
                    <label
                      class="rate__label"
                      for="star-1"
                      title="Ужасно"
                    />
                  </div>
                  <div
                    class="rate__progress"
                  >
                    <span
                      class="rate__stars"
                    >
                      0
                    </span>
                     
                    <span>
                      /
                    </span>
                     
                    <span
                      class="rate__all-stars"
                    >
                      5
                    </span>
                  </div>
                </div>
                <p
                  class="rate__message"
                >
                  Нужно оценить товар
                </p>
              </fieldset>
              <div
                class="custom-input form-review__item "
              >
                <label>
                  <span
                    class="custom-input__label"
                  >
                    Ваше имя
                    <svg
                      aria-hidden="true"
                      height="9"
                      width="9"
                    >
                      <use
                        xlink:href="#icon-snowflake"
                      />
                    </svg>
                  </span>
                  <input
                    name="user-name"
                    placeholder="Введите ваше имя"
                    required=""
                    type="text"
                  />
                </label>
                <p
                  class="custom-input__error"
                >
                  Нужно указать имя
                </p>
              </div>
              <div
                class="custom-input form-review__item "
              >
                <label>
                  <span
                    class="custom-input__label"
                  >
                    Достоинства
                    <svg
                      aria-hidden="true"
                      height="9"
                      width="9"
                    >
                      <use
                        xlink:href="#icon-snowflake"
                      />
                    </svg>
                  </span>
                  <input
                    name="user-plus"
                    placeholder="Основные преимущества товара"
                    required=""
                    type="text"
                  />
                </label>
                <p
                  class="custom-input__error"
                >
                  Нужно указать достоинства
                </p>
              </div>
              <div
                class="custom-input form-review__item "
              >
                <label>
                  <span
                    class="custom-input__label"
                  >
                    Недостатки
                    <svg
                      aria-hidden="true"
                      height="9"
                      width="9"
                    >
                      <use
                        xlink:href="#icon-snowflake"
                      />
                    </svg>
                  </span>
                  <input
                    name="user-minus"
                    placeholder="Главные недостатки товара"
                    required=""
                    type="text"
                  />
                </label>
                <p
                  class="custom-input__error"
                >
                  Нужно указать недостатки
                </p>
              </div>
              <div
                class="custom-textarea form-review__item "
              >
                <label>
                  <span
                    class="custom-textarea__label"
                  >
                    Комментарий
                    <svg
                      aria-hidden="true"
                      height="9"
                      width="9"
                    >
                      <use
                        xlink:href="#icon-snowflake"
                      />
                    </svg>
                  </span>
                  <textarea
                    minlength="5"
                    name="user-comment"
                    placeholder="Поделитесь своим опытом покупки"
                  />
                </label>
                <div
                  class="custom-textarea__error"
                >
                  Нужно добавить комментарий
                </div>
              </div>
            </div>
            <button
              class="btn btn--purple form-review__btn"
              type="submit"
            >
              Отправить отзыв
            </button>
          </form>
        </div>
        <button
          aria-label="Закрыть попап"
          class="cross-btn"
          type="button"
        >
          <svg
            aria-hidden="true"
            height="10"
            width="10"
          >
            <use
              xlink:href="#icon-close"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
